### PR TITLE
Adding click event to controls

### LINKED ISSUE #5872: UcrInput Click Event
**Is your feature request related to a problem? Please describe.**
Some dialogs like `dlgExportRObjects` , `dlgExportRWorkSpace` etc are currently listening to click events of type `ucrInput` control. Currently this event is not being raised because it doesn't exist. The only way to concretely listen for the event is by accessing their inner controls and listening for their click event. 

For instance, the following code may not be raise
```
    Private Sub ucrInput_Click(sender As Object, e As EventArgs) Handles ucrInput.Click
       'do something
    End Sub
```

Instead you may need to rewrite it to this
```
    Private Sub ucrInput_Click(sender As Object, e As EventArgs) Handles ucrInput.txtInput.Click
          'do something
    End Sub
```

This is not a recommended as it goes against the principles of OOP (particularly Encapsulation).

**Describe the solution you'd like**
An extra event needs to be written which enables dialogs to listen to click event of controls of type `ucrInput` without having to access their inner controls.

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes #5872
This PR affects ucrCore, ucrInputTextBox, ucrTextBox controls.

@africanmathsinitiative/developers this is ready for event.
@lloyddewit I added calling of this event to controls that I think currently need it. I was prompted to add this click event when doing PR #5873. I had to remove the code whose intention was to listen for this event but in practice it was not being raised cause technically it didn't exist (I've explained this in the issue). If this is merged before PR #5872 then I will be able to implement the feature that may be needed in that PR correctly.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@africanmathsinitiative/developers I wanted to merge this PR asap. So I peer reviewed and approved it directly. Therefore you do not need to peer review this PR (unless you wish to!). 

@rdstern I don't think there is anything for you to test here.

@maxwellfundi Please could you also review and approve? If everything's OK please could you also merge? thanks

File: instat/ucrInputComboBox.resx [hunk 1]
```diff
@@ -166,10 +166,13 @@
     <value>True</value>
   </metadata>
   <data name="$this.AutoScaleDimensions" type="System.Drawing.SizeF, System.Drawing">
-    <value>6, 13</value>
+    <value>9, 20</value>
+  </data>
+  <data name="$this.Margin" type="System.Windows.Forms.Padding, System.Windows.Forms">
+    <value>9, 12, 9, 12</value>
```
Comment (shadrackkibet) [inline instat/ucrInputComboBox.resx:172]: Out of interest, why is there a change in the resource file? I thought this is just meant to add `Click` event.

File: instat/ucrInputComboBox.resx [hunk 2]
```diff
@@ -122,13 +122,13 @@
   </metadata>
   <assembly alias="System.Drawing" name="System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
   <data name="mnuRightClickCopy.Size" type="System.Drawing.Size, System.Drawing">
-    <value>102, 22</value>
+    <value>126, 32</value>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:125]:
```suggestion
    <value>102, 22</value>
```

File: instat/ucrInputComboBox.resx [hunk 3]
```diff
@@ -122,13 +122,13 @@
...
   </data>
   <data name="mnuRightClickCopy.Text" xml:space="preserve">
     <value>Copy</value>
   </data>
   <data name="mnuRightClick.Size" type="System.Drawing.Size, System.Drawing">
-    <value>103, 26</value>
+    <value>127, 36</value>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:131]:
```suggestion
    <value>103, 26</value>
```

File: instat/ucrInputComboBox.resx [hunk 4]
```diff
@@ -144,7 +144,7 @@
     <value>0, 0</value>
   </data>
   <data name="cboInput.Size" type="System.Drawing.Size, System.Drawing">
-    <value>137, 21</value>
+    <value>206, 28</value>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:147]:
```suggestion
    <value>137, 21</value>
```

File: instat/ucrInputComboBox.resx [hunk 5]
```diff
@@ -166,10 +166,13 @@
     <value>True</value>
   </metadata>
   <data name="$this.AutoScaleDimensions" type="System.Drawing.SizeF, System.Drawing">
-    <value>6, 13</value>
+    <value>9, 20</value>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:169]:
```suggestion
    <value>6, 13</value>
```

File: instat/ucrInputComboBox.resx (hunk 1 above)
Comment (Patowhiz) [reply instat/ucrInputComboBox.resx:172]: Thanks, these were accidental changes. Undoing them now

File: instat/ucrInputComboBox.resx [hunk 6]
```diff
@@ -168,8 +168,11 @@
   <data name="$this.AutoScaleDimensions" type="System.Drawing.SizeF, System.Drawing">
     <value>6, 13</value>
   </data>
+  <data name="$this.Margin" type="System.Windows.Forms.Padding, System.Windows.Forms">
+    <value>9, 12, 9, 12</value>
+  </data>
   <data name="$this.Size" type="System.Drawing.Size, System.Drawing">
-    <value>137, 21</value>
+    <value>206, 32</value>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:175]:
```suggestion
    <value>137, 21</value>
```

File: instat/ucrInputComboBox.Designer.vb [hunk 7]
```diff
@@ -55,6 +55,7 @@ Partial Class ucrInputComboBox
         '
         'mnuRightClick
         '
+        Me.mnuRightClick.ImageScalingSize = New System.Drawing.Size(24, 24)
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.Designer.vb:58]:
```suggestion
       
```

File: instat/ucrInputComboBox.Designer.vb [hunk 8]
```diff
@@ -55,6 +55,7 @@ Partial Class ucrInputComboBox
         '
         'mnuRightClick
         '
+
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.Designer.vb:58]:
```suggestion
```

File: instat/ucrInputComboBox.resx [hunk 9]
```diff
@@ -168,6 +168,9 @@
   <data name="$this.AutoScaleDimensions" type="System.Drawing.SizeF, System.Drawing">
     <value>6, 13</value>
   </data>
+  <data name="$this.Margin" type="System.Windows.Forms.Padding, System.Windows.Forms">
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:171]:
```suggestion
```

File: instat/ucrInputComboBox.resx [hunk 10]
```diff
@@ -168,6 +168,9 @@
...
+    <value>9, 12, 9, 12</value>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:172]:
```suggestion
```

File: instat/ucrInputComboBox.resx [hunk 11]
```diff
@@ -168,6 +168,9 @@
...
+  </data>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:173]:
```suggestion
```

File: instat/ucrInputComboBox.resx [hunk 12]
```diff
@@ -183,4 +183,4 @@
   <data name="&gt;&gt;$this.Type" xml:space="preserve">
     <value>instat.ucrInput, instat, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null</value>
   </data>
-</root>
\ No newline at end of file
+</root>
```
Comment (Patowhiz) [inline instat/ucrInputComboBox.resx:186]:
```suggestion
</root>
```

**Comment by Patowhiz** [comment]
Removed all the accidental resource and design changes. Ready for review

**Comment by lloyddewit** [comment]
@shadrackkibet Thanks for questioning the resource file changes.
@maxwellfundi Please could you also review and approve? If everything's OK please could you also merge? thanks

**Comment by lloyddewit** [comment]
@rdstern I understand that this PR is blocking PR #5872 and PR #5880. There is nothing for you to test here but please could you approve anyway so that we can merge? thanks

**Pull request merged**